content/src/instrument.ts

```diff
@@ -0,0 +1,24 @@
+import { AppComponents } from './types'
+
+export function instrumentWithQueryDurationMetric<F extends (...args: any[]) => any>(
+  components: Pick<AppComponents, 'metrics'>,
+  queryNameLabel: string,
+  functionToInstrument: F
+): F {
```

Comment:
May this be simplified? It may reduce the final code using closures

```diff
      ? await instrumentWithQueryDurationMetric(
          components,
          durationQueryNameLabel,
-         (sql: SQLStatement): Promise<IDatabase.IQueryResult<T[]>> => {
+         async () => {
            return pool.query<T[]>(sql)
          }
-       )(sql)
+       )
```

```suggestion
export function instrumentWithQueryDurationMetric<T>(
  components: Pick<AppComponents, 'metrics'>,
  queryNameLabel: string,
  functionToInstrument: () => Promise<T>
): Promise<T> {
```

Reply:
Yes, sure. I don't know why I thought it was a better idea to wrap the function only, for re-usage maybe. Anyways, refactored.